DSharpPlus.Commands/Invocation/CommandEmitUtil.cs

```diff
@@ -19,106 +19,122 @@ internal static class CommandEmitUtil
 
-        EmitMethodWrapper(il, method);
+        // Create the wrapper logic
+        EmitMethodWrapper(dynamicMethod.GetILGenerator(), method);
 
+        // Return the delegate for the wrapper which invokes the method
         return dynamicMethod.CreateDelegate<Func<object?, object?[], ValueTask>>();
     }
 
-    // get a ValueTask func for a command returning Task, emitting an uniform Task wrapper and then
-    // returning a lambda awaiting that Task wrapper
+    /// <summary>
+    /// Emits a wrapper function around a command method that returns a <see cref="Task"/>.
+    /// </summary>
+    /// <param name="method">The method to wrap.</param>
+    /// <returns>An asynchronous function that wraps the command method, returning a <see cref="ValueTask"/>.</returns>
     private static Func<object?, object?[], ValueTask> GetTaskFunc(MethodInfo method)
     {
-        DynamicMethod dynamicMethod = new
-        (
-            $"{method.Name}-task-wrapper",
-            typeof(Task),
-            [typeof(object), typeof(object?[])]
-        );
-
-        ILGenerator il = dynamicMethod.GetILGenerator();
+        // Create the wrapper function
+        DynamicMethod dynamicMethod = new($"{method.Name}-task-wrapper", typeof(Task), [typeof(object), typeof(object?[])]);
 
-        EmitMethodWrapper(il, method);
+        // Create the wrapper logic
+        EmitMethodWrapper(dynamicMethod.GetILGenerator(), method);
 
+        // Return the delegate for the wrapper which invokes the method
         Func<object?, object?[], Task> taskWrapper = dynamicMethod.CreateDelegate<Func<object?, object?[], Task>>();
 
+        // Create an async wrapper around the task wrapper
         return async (object? instance, object?[] parameters) => await taskWrapper(instance, parameters);
     }
 
-    // the heavy lifting of emitting IL actually just works entirely the same for both, since we don't do
-    // anything further with the returned Task/ValueTask, and our wrapper is synchronous.
-    private static void EmitMethodWrapper(ILGenerator il, MethodInfo method)
+    /// <summary>
+    /// Writes the body of the wrapper function which invokes the command method.
+    /// </summary>
+    /// <param name="dynamicMethodIlGenerator"></param>
+    /// <param name="method"></param>
+    private static void EmitMethodWrapper(ILGenerator dynamicMethodIlGenerator, MethodInfo method)
     {
         ParameterInfo[] parameters = method.GetParameters();
-        ConstructorInfo argumentExceptionCtor = typeof(ArgumentException).GetConstructor([typeof(string), typeof(string)])!;
-
         if (!method.IsStatic)
         {
-            il.Emit(OpCodes.Ldarg_0);
-
-            // not static implies declared within a type
+            // Load the instance (this) onto the stack.
+            dynamicMethodIlGenerator.Emit(OpCodes.Ldarg_0);
             if (method.DeclaringType!.IsValueType)
```

Comment:
removing newlines added for logical separation of logically distinct code is generally ill-advisable